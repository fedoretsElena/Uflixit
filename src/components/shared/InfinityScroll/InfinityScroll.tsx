import React, { Component } from "react";

interface IProps {
    listenUntil: boolean;
    endPageHappend: () => void;
    distanceToEndOfPage?: number;
}

class InfinityScroll extends Component<IProps> {
    render() {
        return <></>;
    }

    componentDidMount(): void {
        this.listenScroll();
    }

    listenScroll(): void {
        const { body } = document;
        const { listenUntil, endPageHappend } = this.props;
        const distanceToEndOfPage = this.props.distanceToEndOfPage || 300;

        window.addEventListener("scroll", () => {
            if (
                window.innerHeight + window.scrollY + distanceToEndOfPage >=
                    body.offsetHeight &&
                listenUntil
            ) {
                endPageHappend();
            }
        });
    }
}

export default InfinityScroll;
