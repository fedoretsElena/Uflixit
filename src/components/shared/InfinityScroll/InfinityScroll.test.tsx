import InfinityScroll from "./InfinityScroll";

describe("InfinityScroll component", () => {
    it("should renders correctly", () => {
        const component = shallow(
            <InfinityScroll
                listenUntil={false}
                endPageHappend={() => jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
