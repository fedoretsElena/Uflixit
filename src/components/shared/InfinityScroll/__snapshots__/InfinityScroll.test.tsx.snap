// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfinityScroll component should renders correctly 1`] = `<Fragment />`;
